package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,105 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
+class ParallelExecutor(object):
```

Comment:
```suggestion
class ParallelExecutor:
```
I don't think this is necessary in a Py3+ world anymore.